### ISSUE TITLE
Optimize hierarchy method linkage

### ISSUE THREAD

**Issue by Col-E** [Col-E/Recaf#67]
## Optimize hierarchy method linkage

As of right now when using the `Linked method renaming` feature, loading inputs can range from very fast to very slow *(Multi-threading helps combat this, but not everyone can just throw a dozen threads at the task)*.

The process roughly is:

1. For every class in the input, link parent/child relations
2. For every method in every class, check if linkage has occurred
    * If no linkage, iterate the method's owner's _(class)_ children and parents and search for methods matching the name and descriptor. Link these two together. 
        * The process allows self linkage + methods are linked in both the `parents` and `children` list
        * These are bugs, but do not break the overall functionality so are lower priority bugs
    * If method has been linked already, move on to the next method to be linked.

I'm sure there are some tricks that could be done to skip useless computations. When using this process on Recaf _(7.63 MB jar, 4732 classes)_ it takes about 12 seconds. Ideally the process would take 1 or 2, but for this size of input I'd be happy with a 50% reduction.

**Comment by Col-E** [comment]
90a3c9a5c32d2eb06ca1c341f83affd2f03ef39c Not optimization, but moving it to a separate thread makes it less intrusive

**Comment by Col-E** [comment]
a44dfbc498924b5e8c15cbdda9c2be21631135b1 Was not intended to be an optimization but the different approach not only resolved most _(If not all)_ of the bugs and is certainly _much_ faster. 

* **Before**: Loading Recaf.jar _(7.63 MB jar, 4732 classes)_  -> about 12 seconds
* **After**: Loading Recaf.jar _(7.63 MB jar, 4732 classes)_  -> about 0.15 seconds

About 80x faster.... Well thats certainly an improvement. I didn't even bother to re-add threading so it could go a bit faster after throwing some extra CPU cores at the task.
